services/QuillLMS/lib/tasks/archive_classroom_units.rake

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+namespace :archived_classrooms do
+  desc 'Archive classroom units for archived classrooms using parallel workers'
+  task :archive_units, [:batch_size] => :environment do |task, args|
+    batch_size = (args[:batch_size] || 50).to_i
+
+    puts 'Starting parallel job to archive classroom units for archived classrooms...'
+    puts "Using batch size: #{batch_size}"
+
+    scope = Classroom.unscoped
+      .where(visible: false)
+      .where(
+        ClassroomUnit
+          .where('classroom_units.classroom_id = classrooms.id')
+          .where(visible: true)
+          .arel.exists
+      )
+      .order(created_at: :desc)
+
+    total_count = scope.count
+    puts "Found #{total_count} archived classrooms with visible classroom units"
+
+    if total_count == 0
+      puts 'No archived classrooms found to process'
+      next
+    end
+
+    classroom_ids = scope.pluck(:id)
+    batches = classroom_ids.each_slice(batch_size).to_a
+
+    puts "Enqueueing #{batches.size} jobs..."
+
+    batches.each_with_index do |batch_ids, index|
+      ArchiveClassroomUnitsWorker.perform_async(batch_ids)
+      puts "Enqueued batch #{index + 1}/#{batches.size} with #{batch_ids.size} classrooms"
+    end
+
+    puts 'All jobs enqueued! Monitor progress in Rails logs or Sidekiq web UI at /sidekiq'
+    puts 'Usage: bundle exec rake archived_classrooms:archive_units[200] to set batch size to 200'
```

Comment:
it's better practice to put usage messages at the start of a script (can therefore return early)